springfox-core/src/main/java/springfox/documentation/builders/OperationBuilder.java

```diff
@@ -232,9 +233,20 @@ public OperationBuilder tags(Set<String> tags) {
     return this;
   }
 
+  /**
+   * Sets the nickname for the operation
+   *
+   * @param nickname - a string for the nickname
+   * @return this
+   */
+  public OperationBuilder nickname(String nickname) {
+      this.nickname = defaultIfAbsent(nickname, this.nickname);
+      return this;
+  }
+
   public Operation build() {
     return new Operation(method, summary, notes, responseModel, uniqueId, position, tags, produces,
-        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden);
+        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden, nickname);
```

Comment:
@adrianbk I dont think we need to introduce an operation id generator concept. Thats what the plugin model is for, there is an operation builder plugin to enrich the operation model already. So l think the direction @who has taken would be the best approach. 

Just so we dont over engineer it, its probably fine to generate a operation id regardless. The extensibility mechanism allows folks to override the name if they wish.